File: selfstudy/golang/apples_reverse.go
/*
task 208:
Apples - Reverse
You need to reverse the order of apple weights in grams
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderApples(apples []int) []int {
	res := make([]int, len(apples))
	for i, v := range apples {
		res[len(apples)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderApples([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderApples([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderApples([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderApples([]int{100, 200})},
		{[]int{}, Reverse_orderApples([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderApples(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/apples_reverse.js
/*
task 8:
Apples - Reverse
You need to reverse the order of apple weights in grams
*/

function reverse_order_apples(apples) {
  return [...apples].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_apples([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_apples([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_apples([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_apples([100, 200]) },
  { input: [], expected: reverse_order_apples([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_apples', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_apples(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_apples(t.input) === t.expected || JSON.stringify(reverse_order_apples(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done
